pkg/synchromanager/clustersynchro_manager.go

```diff
@@ -256,6 +271,17 @@ func (manager *Manager) processNextCluster() (continued bool) {
 
 // if err returned is not nil, cluster will be requeued
 func (manager *Manager) reconcileCluster(cluster *clusterv1alpha2.PediaCluster) controller.Result {
+	if cluster.Status.ShardingName == nil && cluster.Spec.ShardingName != manager.shardingName {
+		return controller.NoRequeueResult
+	}
+
+	if cluster.Status.ShardingName != nil && *cluster.Status.ShardingName != manager.shardingName {
+		return controller.NoRequeueResult
+	}
```

Comment:
I think this could be put into a separate function, which would add more comprehensive unit testing to it.
But it can be optimized in a new pr.